File: 6_multilevel_inheritance.py
# Base class super class
class A:
    def method_a(self):
        print("This is method from class A")

# Intermediate class (Parent class)
class B(A):
    def method_b(self):
        print("This is method from class B")

# Derived class (Child class)
class C(B):
    def method_c(self):
        print("This is method from class C")

# Creating an instance of the child class C
obj = C()

# Calling methods from all three classes
obj.method_a()  # Inherited from class A
obj.method_b()  # Inherited from class B
obj.method_c()  # Defined in class C

File: 6_multilevel_inheritance.cpp
#include <iostream>
using namespace std;

// Base class (Grandparent class)
class A {
public:
    void methodA() {
        cout << "This is method from class A" << endl;
    }
};

// Intermediate class (Parent class)
class B : public A {
public:
    void methodB() {
        cout << "This is method from class B" << endl;
    }
};

// Derived class (Child class)
class C : public B {
public:
    void methodC() {
        cout << "This is method from class C" << endl;
    }
};

int main() {
    // Creating an object of class C
    C obj;

    // Calling methods from all three classes
    obj.methodA();  // Inherited from class A
    obj.methodB();  // Inherited from class B
    obj.methodC();  // Defined in class C

    return 0;
}
